handel.go

```diff
@@ -279,20 +283,37 @@ func (h *Handel) Stop() {
 func (h *Handel) periodicUpdate() {
 	for i := byte(1); i <= byte(len(h.levels)); i++ {
 		lvl := h.getLevel(i)
-		if !lvl.sendStarted {
-			h.decideToStartLevel(lvl)
-		}
 		if lvl.active() {
-			h.sendUpdate(lvl, 1)
+			h.sendUpdate(lvl, h.c.UpdateCount)
 		}
 	}
 }
 
-func (h *Handel) decideToStartLevel(l *level) {
-	msSinceStart := int(time.Now().Sub(h.startTime).Seconds() * 1000)
-	if msSinceStart >= l.id*int(h.c.LevelTimeout.Seconds())*1000 {
-		l.sendStarted = true
+// StartLevel starts the given level if not started already. It sends
+// our best signature for this level up to CandidateCount peers.
+func (h *Handel) StartLevel(level int) {
+	h.Lock()
+	defer h.Unlock()
+	lvl := h.getLevel(byte(level))
+	if lvl.started() {
+		return
+	}
+	lvl.setStarted()
+	h.sendUpdate(lvl, h.c.CandidateCount)
+}
+
+// Send our best signature set for this level, to 'count' nodes
+func (h *Handel) sendUpdate(l *level, count int) {
+	if !l.started() {
+		panic("level not started!")
+	}
+	if !l.active() {
```

Comment:
I don't really like the idea to have two different cases here:
- error when not started
- not an error when not active

I think that the previous logic was better (always panic)
As well I'm not sure that it's not better to embed the timeout logic in the active one (i.e. if a level has timeouted then it's active)